src/logics/ArithLogic.cc

```diff
@@ -351,6 +351,33 @@ namespace {
         return logic.mkPlus(std::move(args)); // TODO: Can we use non-simplifying versions of mkPlus/mkTimes?
     }
 
+    PTRef polyToPTRefSubstitution(ArithLogic & logic, PTRef var, poly_t & poly) {
+        if ((logic.hasUFs() or logic.hasArrays()) and logic.isVar(var)) {
+            if (std::ranges::any_of(poly, [&logic](auto const & term) {
+                    return term.var != PTRef_Undef and not logic.isVar(term.var);
+                })) {
+                return PTRef_Undef;
+            }
+        }
+
+        Real coeff = poly.getCoeff(var);
+        bool const isMinusOne = coeff.isOne();
+        coeff.negate();
+        bool const isOne = coeff.isOne();
```

Comment:
While this is the most efficient way to determine these flags, it is also very confusing.

On the first read, the line `bool const isMinusOne = coeff.isOne()` immediately raised a red flag in my mind.
Only after thinking about it for a while, I realized what is actually going on.

Please also add a note that these boolean flags refer to the state when the variable is put on the other side of equality (and therefore these are properties of the coefficient only **after** it has been negated).